anchore_engine/services/policy_engine/engine/vulns/mappers.py

```diff
@@ -922,10 +922,17 @@ def to_engine_vulnerabilities(self, grype_vulnerabilities):
             if grype_vulnerability:
 
                 # Transform the versions block
-                if grype_vulnerability.deserialized_fixed_in_versions:
-                    version = ",".join(
-                        grype_vulnerability.deserialized_fixed_in_versions
-                    )
+                if grype_vulnerability.version_constraint:
+                    if "unspecified" in grype_vulnerability.version_constraint.lower():
+                        version = "*"
```

Comment:
@Vijay-P The inference behind using `*` is that all versions of the package are vulnerable i.e. the grype would match any version of this package to the vulnerability. I'm not aware of any such grype matching logic that special cases "Unspecified" version. It's possible I could be missing something. Otherwise the PR looks good to me

Reply:
Sorry, I missed this before

Reply:
No problem. I only implemented this because i found data in the grype DB that follows this pattern. After taking a second look at the data with this query:
```
SELECT id, namespace, version_constraint, fixed_in_versions, fix_state FROM vulnerability WHERE vulnerability.version_constraint LIKE '%unspecified%' OR vulnerability.version_constraint LIKE '%unknown%';
```
it has become apparent that the "unspecified" check that I used cannot be broadly applied, as while most of them are not fixed, a couple of records have a fixed-in version.